watertap/core/tests/test_zero_order_base.py

```diff
@@ -0,0 +1,341 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Tests for general zero-order property package
+"""
+import pytest
+
+from idaes.core import declare_process_block_class, FlowsheetBlock
+from idaes.core.util.exceptions import ConfigurationError
+from pyomo.environ import ConcreteModel, Var
+
+from watertap.core import ZeroOrderBaseData
+from watertap.core import WaterParameterBlock
+import idaes.logger as idaeslog
+
+
+@declare_process_block_class("DerivedZOBase")
+class DerivedZOBaseData(ZeroOrderBaseData):
+    def build(self):
+        super().build()
+
+
+@pytest.mark.unit
+def test_private_attributes():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.params = WaterParameterBlock(default={"solute_list": ["A", "B", "C"]})
+
+    m.fs.unit = DerivedZOBase(default={"property_package": m.fs.params})
+
+    assert m.fs.unit._tech_type is None
+    assert m.fs.unit._has_recovery_removal is False
+    assert m.fs.unit._fixed_perf_vars == []
+    assert m.fs.unit._initialize is None
+    assert m.fs.unit._scaling is None
+    assert m.fs.unit._get_Q is None
```

Comment:
Why do we need `_get_Q`? I noticed that the unit models will have there own `get_Q` method that just grabs the inlet flowrate, but I didn't see how it was necessary at this point. Was this done in preparation for shifting to a mass flow basis in the property model in a subsequent PR?

Reply:
This is needed because pass-through units on have one StateBlock named `properties` - this method is there to get around the different names.